### ISSUE TITLE
Customize note search results filtering

### ISSUE THREAD

**Issue by archwa** [GitJournal/GitJournal#648]
# Feature Description

As a user, I would like to filter note search results according to *only some of* note title, note filename, and note body.

## Context

Currently, note search results filtering is performed by matching a query to *any of* note title, note filename, and note body. I believe this is accomplished in the note search delegate. [See here](https://github.com/GitJournal/GitJournal/blob/master/lib/widgets/note_search_delegate.dart#L69) for implementation details.

## Examples

For example, I want to search by note filename alone. When I provide a query containing a part of the filename that is also included in the title or in the body of other note(s), I may receive a number of undesirable results first. Consequently, it is more difficult to find a note with a particular filename.

Custom filtering could be implemented using check boxes, like this:
Match query to note
- [ ] Title
- [x] Filename
- [ ] Body

... for example.

Or, custom filtering could be implemented using a reorderable priority sort on the search results, listing results first by (1), then by (2), finally by (3):
1. Filename
2. Title
3. Body

... for example. Not a pure filtering solution, but accomplishes a similarly desirable result.

These are just two implementation ideas -- I could think of a few other ways to do this.

## Final Thoughts

The ability to quickly search and filter notes based on query matches to *only some of* note title, note filename, and note body will make note searching much easier, and it will provide tremendous value to myself and others who use GitJournal now and in the future.

**Comment by m4rz910** [comment]
Agreed I would benefit from this feature. I use dendron on my computer and the hierarchical search there works on file names. I often struggle to find the note I want on gitjournal because theres so many other notes returned when I want to simply search by file name.
